### ISSUE TITLE
metalink :: get (or even put) like option

### ISSUE THREAD

**Issue by adriansev** [xrootd/xrootd#1048]
HI! It would be nice if xrdcp (and python addJob) would have an option for specifying as single argument the meta4 file and infer from it's contents the destination  (write to the name= field the url= sources). this would simplify the download from generated metafiles (xrdcp -getmeta file.meta4) and would allow downloading of collections of files (by iterating over <file></file> groups in the meta4 collection).
Also, it could be very interesting to have a -putmeta like option where the name= field from meta is used as source and urls are the destination

**Comment by adriansev** [comment]
no traction, this can be done in a python wrapper

**Comment by amadio** [comment]
Indeed, the current implementation only supports a single file. This is indeed something that makes sense to implement in the future. Maybe bring this up in the collaboration meeting?

**Comment by adriansev** [comment]
> Indeed, the current implementation only supports a single file. This is indeed something that makes sense to implement in the future. Maybe bring this up in the collaboration meeting?

@amadio well, i'm not sure if would be useful for everyone .. for a metalink user i would say that it is useful, as one could create the metalink with all relevant information (src and dst) and then just add the metalink as is to copyprocess (or xrdcp),  that in turn, would extract all needed information from there.. (so both downloads and upload many replicas would be easier to do) but this is something that for myself i already covered ..

**Comment by amadio** [comment]
Well, at least we should consider bringing our implementation closer to [RFC5854](https://datatracker.ietf.org/doc/html/rfc5854), which does allow what you describe. If you use a metalink per file with XRootD, setting the name of the output, at least, works.

**Comment by amadio** [comment]
Hi, @adriansev I actually meant that you can set the output name in the metalink (see https://datatracker.ietf.org/doc/html/rfc5854#section-4.1.2.1). Then you can copy the metalink to `.` and get the correct output name.

**Comment by adriansev** [comment]
@amadio Thanks a lot! a bit annoying that the paths are only relative but i will try to see if i can `cd /` and then do the copy to whatever destination ..

**Comment by adriansev** [comment]
@amadio i just try it out and with:
```
grep name a51312ba-62af-56fc-90da-9f58a29c54ca.meta4 
   <file name="bla/bla2/new_name.sh">
```
doing `xrdcp -PNf -v a51312ba-62af-56fc-90da-9f58a29c54ca.meta4 .`
i got `new_name.sh` in current directory, in opposition to:
```
Directory information can also be contained in a "path/file" format
   only, as in:

   <file name="debian-amd64/sarge/Contents-amd64.gz">

   In this example, a subdirectory "debian-amd64/sarge/" will be created
   and a file named "Contents-amd64.gz" will be created inside it.

      Security Note: The path MUST NOT contain any directory traversal
      directives or information.  The path MUST be relative.  The path
      MUST NOT begin with a "/", "./", or "../"; contain "/../"; or end
      with "/..".
```

my wish would have been to specify a full path in name and be able to not use a destination .. 
so with something like  `xrdcp -PNf -v a51312ba-62af-56fc-90da-9f58a29c54ca.meta4` the destination is set to the `name` property of `file` .. 
and yes, the extension to multiple files in metalink would be wonderful as it would allow me for recursive or list of files copy to just create a single metalink and then pass it to copyprocess as a single job but i'm not sure how it would go with `CopyProcessHandler` where i do printouts per job in both `begin` and `end` ... anyway this seems a lot of work with not clear benefit so this can be kept closed until more need for this appear.
Thanks a lot!!!

**Comment by amadio** [comment]
Ah, indeed, when I tried this feature myself, I didn't add subdirectories to the output path name. This is probably something quite easy to add. As for multiple file support, it does require more work, but eventually I'd like to add support as well, if only to conform better to the RFC, but this has low priority relative to other development tasks.
